projects/plugins/social/src/class-share-limits.php

```diff
@@ -69,6 +82,19 @@ public function has_more_shares_than_connections() {
 	 * Render a notice with the share count in the classic editor.
 	 */
 	public function render_classic_editor_notice() {
+		$current_page_url = null;
+		if ( isset( $_SERVER['REQUEST_SCHEME'] ) && isset( $_SERVER['HTTP_HOST'] ) && isset( $_SERVER['REQUEST_URI'] ) ) {
+			// phpcs:ignore -- WordPress.Security.ValidatedSanitizedInput.InputNotSanitized If we use esc_raw_url, it adds http to the beginning.
+			$current_page_url =  wp_unslash( $_SERVER['REQUEST_SCHEME'] ) . '://' . wp_unslash( $_SERVER['HTTP_HOST'] ) . wp_unslash( $_SERVER['REQUEST_URI'] );
+		}
+		$redirect_url = $current_page_url ? Redirect::get_url(
+			'jetpack-social-basic-plan-classic-editor',
+			array(
+				'query' => 'redirect_to=' . $current_page_url,
+			)
+		) :
+		Redirect::get_url( 'jetpack-social-basic-plan-classic-editor' );
```

Comment:
You could make the the query parameters the ternary instead:

`$params = $current_page_url ? array( 'query' => 'redirect_to=' . $current_page_url ) : array()`

That being said, I think it would be good to always default to `admin_url` for redirecting, rather than not redirecting at all.

Reply:
Good suggestion, will do it